pkg/controller/worker/machines.go

```diff
@@ -171,6 +176,7 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 				"machineType":      pool.MachineType,
 				"keyName":          infrastructureStatus.Node.KeyName,
 				"networkID":        infrastructureStatus.Networks.ID,
+				"subnetID":         subnet.ID,
```

Comment:
Do we need to pass the `subnetID` always? I mean we need it indeed for Shoots in existing networks. But what about all other Shoots with are currently deployed into Gardener managed networks. Does it have an impact on the existing Shoots when we now always add the `subnetID` to their machine classes?